nuxeo-features/nuxeo-platform-oauth/src/main/java/org/nuxeo/ecm/platform/oauth2/clients/OAuth2ClientServiceImpl.java

```diff
@@ -70,28 +71,72 @@ public OAuth2Client getClient(String clientId) {
         return queryClients().stream().map(OAuth2Client::fromDocumentModel).collect(Collectors.toList());
     }
 
+    @Override
+    public OAuth2Client create(OAuth2Client oAuth2Client) {
+        Objects.requireNonNull(oAuth2Client, "oAuth2Client is required");
+        if (getClientModel(oAuth2Client.getId()) != null) {
+            throw new NuxeoException(String.format("Client with id '%s' already exist", oAuth2Client.getId()),
+                    SC_BAD_REQUEST);
+        }
+
+        return execute((session) -> {
```

Comment:
According to the Sonar rule `squid:S1611`, you should remove the parentheses around the `session` parameter.